### DISCUSSION TITLE
[ISSUE]: Error when upgrading from 5.x to 6: Requested value 'Mainline' was not found.

### DISCUSSION THREAD

**Discussion by trondhindenes** [GitTools/GitVersion#4132] [Q&A] [answered]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

gittools/actions

### GitVersion version

6.0

### Operating system

Linux

### What are you seeing?

We're using trying to move from the `gittools/gitversion:5.12.0-ubuntu.20.04-6.0` docker image to the newest one `gittools/gitversion:6.0.0-ubuntu.24.04-8.0`. However, we're getting this exception when running our builds with the newest image: 
```
Could not build the configuration instance because following exception occurred: 'Requested value 'Mainline' was not found.' Please ensure that the /overrideconfig parameters are correct and the configuration file is in the correct format.
```

The GitVersion.yml file in the repo in question is very simple:
```
---
mode: Mainline

```

This has worked well for us for years using 5.x, is there anything we need to do in order to transition to 6.0?

Btw the dropdown "GitVersion package" options doesn't include "docker image". We're not using "gittools/actions" but I had to put something in there.

### What is expected?

I was hoping the build would just work.

### Steps to Reproduce

See issue description

**Comment by arturcic** [comment]
@HHobeck can you help here with the correct configuration?

**Comment by HHobeck** [comment, marked as answer]
In the breaking changes documentation the following is written:

> The Mainline mode and the related implementation has been removed completely. The new Mainline version strategy should be used instead.

and in the configuration documentation:

> The following supported workflow configurations are available in GitVersion and can be referenced by the workflow property:
> 
> GitFlow (GitFlow/v1)
> GitHubFlow (GitHubFlow/v1)
> TrunkBased (TrunkBased/preview1)

The preview built-in configuration (experimental usage only) for the TrunkBased workflow uses the Mainline version strategy:

```yaml
workflow: TrunkBased/preview1
```

Alternatively, you can use the GitHubFlow workflow with mainline version strategy as follows (experimental use only):
```yaml
workflow: GitHubFlow/v1
strategies:
- ConfiguredNextVersion
- Mainline
```

Link 1: [BREAKING_CHANGES.md](https://github.com/GitTools/GitVersion/blob/main/BREAKING_CHANGES.md)
Link 2: [configuration.md](https://github.com/GitTools/GitVersion/blob/main/docs/input/docs/reference/configuration.md)

**Reply by trondhindenes** [reply]
Thanks. I looked for something in https://gitversion.net/docs/ but couldn't find anything related to moving from 5.x to 6.x. Maybe it would make sense to link to the breaking changes markdown in the git repo, since it's quite important.

**Reply by asbjornu** [reply]
We should definitely add a "breaking changes" page to the docs, @trondhindenes. Pull requests are welcome. :)
